src/CircuitSimulator.cpp

```diff
@@ -83,6 +85,18 @@ std::map<std::string, std::size_t> CircuitSimulator<Config>::simulate(std::size_
     return measurementCounter;
 }
 
+template<class Config>
+dd::fp CircuitSimulator<Config>::expectationValue(const qc::QuantumComputation& observable) {
+    // simulate the circuit to get the state vector
+    simulate(0);
```

Comment:
You might want to call `singleShot` directly, with `simulate(0)` you may end up not running the simulation altogether. (Or just call `simulate(1)`).

Reply:
Thanks again for the suggestion. Makes a lot of sense.
Done in https://github.com/cda-tum/ddsim/pull/203/commits/c084051726b89460037e599f19450a9899f0718d.